code/script.js

```diff
@@ -29,23 +31,155 @@ const showMessage = (message, sender) => {
+  showMessage("Bonjour developer student, what's your name?", 'bot')
+}
+
+//Global setTimeout for the Greeting
+setTimeout(greetUser, 800)
+
+
+//When name is input by user.
+const handleNameInput = event => {
+  event.preventDefault();
+
+  if (run) {
+    userName = nameInput.value;
+    nameInput.value = '';
+    showMessage(`${userName}`, "user");
+    setTimeout(reply, 1000);
+  }
+}
+
+nameForm.addEventListener('submit', handleNameInput);
+
+//Riddles start here
+const reply = () => {
+  showMessage(`I think you need a riddle to relax your mind ${userName}. Choose a riddle by clicking on one of the apples below:`, 'bot');
+  //Buttons for the ridddles
+  nameForm.innerHTML = `
+      <button class="options" id='oneBtn'>🍏</button>
+      <button class="options" id='twoBtn'>🍎</button>
+    `
+  document.getElementById('oneBtn')
+    .addEventListener('click', () => riddleOption('one'))
+
+  document.getElementById('twoBtn')
+    .addEventListener('click', () => riddleOption('two'))
+}
+
+//Different options of riddles that occurs depending on apple-choice. Conditional statements.
+const riddleOption = option => {
+  //Riddle choice: green apple
+  if (option === "one") {
+    showMessage('🍏', 'user')
+    showMessage('🍏: I am used for styling, look and feel. Colors and layout make it real. In files separate, my rules align. Junior dev, what am I, in web design?', 'bot')
+    //Buttons for green apple riddle
+    nameForm.innerHTML = `
+    <button class="options" id='htmlBtn'>HTML</button>
+    <button class="options" id='cssBtn'>CSS</button>
+  `
+
+    document.getElementById('htmlBtn')
+      .addEventListener('click', () => greenWrongAnswer('greenHtml'))
+
+    document.getElementById('cssBtn')
+      .addEventListener('click', () => greenCorrectAnswer('greenCss'))
+  }
+  //Riddle choice: red apple
+  else if (option === "two") {
+    showMessage('🍎', 'user')
+    showMessage('🍎: I bring interactivity, make things fun. With functions and events, the job has begun. In the browser realm, where actions play smart. Junior dev can you name this part?', 'bot')
+    //Buttons for red apple riddle
+    nameForm.innerHTML = `
+      <button class="options" id='jsBtn'>JavaScript</button>
+      <button class="options" id='htmlBtn'>HTML</button>
+    `
+
+    document.getElementById('jsBtn')
+      .addEventListener('click', () => redCorrectAnswer('redJs'))
+    document.getElementById('htmlBtn')
+      .addEventListener('click', () => redWrongAnswer('redHtml'))
+
+  }
+}
+
+//Depending on the answer of the riddles, one of these four functions will start:
+//Wrong answer green apple riddle
+const greenWrongAnswer = () => {
+  showMessage('HTML', 'user')
+  showMessage('Almost! Correct answer is CSS. Thanks for playing with me! Good luck on your studies, you are awesome!', 'bot')
+  showMessage('<img src="https://cdn.pixabay.com/photo/2023/07/13/08/26/ai-generated-8124187_1280.jpg" alt="Happy dog" style="width: 225px; height: auto;" />', 'bot');
```

Comment:
I really love that you put an images inside the end messages, it made me smile after talking to your bot :)